apps/tickets/pay.py

```diff
@@ -108,13 +108,30 @@ def pay(flow="main"):
 
     form.basket_total.data = basket.total
 
+    # Whether the user has an admission ticket in their basket or already purchased.
+    # FIXME: this is rather ugly
+    has_admission_ticket = any(
+        (
+            p.product.is_adult_ticket()
+            and p.state not in ("cancelled", "refunded", "reserved")
```

Comment:
I agree this is a fiddly, but on the plus side it's only needed in this part of the process. Maybe clearer to do `p.state in ("paid", "payment-pending")`? (states at https://github.com/emfcamp/Website/blob/main/models/purchase.py#L13-L17)